.github/actions/build-ab/index.js

```diff
@@ -9,84 +10,131 @@ import { fetchRetry } from '../shared-utils/fetch-retry.js'
 import Handlebars from 'handlebars'
 
 const __dirname = path.dirname(url.fileURLToPath(import.meta.url))
-const template = Handlebars.compile(await fs.promises.readFile(path.resolve(__dirname, './template.js'), 'utf-8'))
-
-let nextScript
-const abScripts = []
+const outputDir = path.resolve(__dirname, '../../../temp')
+const fileList = []
 
 // 0. Ensure the output directory is available and the target file does not exist
+await fs.promises.mkdir(outputDir, { recursive: true })
 
-const outputFile = path.join(
-  path.resolve(__dirname, '../../../temp'),
-  `${args.environment}.js`
-)
-await fs.promises.mkdir(path.dirname(outputFile), { recursive: true })
-if (fs.existsSync(outputFile)) {
-  await fs.promises.rm(outputFile)
+// 1. Create the released environment script
+const releasedScriptRequest = await fetchRetry(`${args.released}?_nocache=${uuidv4()}`, { retry: 3 })
+
+if (!releasedScriptRequest.ok() || typeof releasedScriptRequest.text !== 'function') {
```

Comment:
if fetchRetry doesn't return the fetch promise (which i think is what happens if it fails to load 3 times), .ok() will not be a function and that will probably be the error that throws

something like `.ok is not a function`

Reply:
Fixed